File: jvm/src/main/kotlin/org/faziodev/saywhatstarter/types/QueryResult.kt
package org.faziodev.saywhatstarter.types

import org.faziodev.saywhatstarter.types.messages.Message

data class QueryResult(
    val queryText: String,
    val action: String? = "",
    val parameters: Map<String, String>,
    val allRequiredParamsPresent: Boolean,
    val fulfillmentText: String,
    val fulfillmentMessages: List<Message>,
    val outputContexts: List<Context>?,
    val intent: Intent,
    val intentDetectionConfidence: Double,
    val diagnosticInfo: Map<String, String>?,
    val languageCode: String
)

File: csharp/SayWhatStarterWebhook/Models/QueryResult.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using SayWhatStarterWebhook.Converters;
using SayWhatStarterWebhook.Models.Messages;

namespace SayWhatStarterWebhook.Models
{
    public class QueryResult
    {
        public string QueryText { get; set; }
        public string Action { get; set; }
        public IDictionary<string, string> Parameters { get; set; }
        public bool AllRequiredParamsPresent { get; set; }
        public string FulfillmentText { get; set; }
        [JsonProperty(ItemConverterType = typeof(MessageConverter))]
        public IList<IMessage> FulfillmentMessages { get; set; }
        public IList<Context> OutputContexts { get; set; }
        public Intent Intent { get; set; }
        public double IntentDetectionConfidence { get; set; }
        public IDictionary<string, string> DiagnosticInfo { get; set; }
        public string LanguageCode { get; set; }
    }
}